api/batches/hooks.js

```diff
@@ -99,29 +100,38 @@ Batches.after.update(
     [Games, GameLobbies].forEach(coll => {
       coll.update({ batchId }, { $set: { status } }, { multi: true });
     });
-  },
-  { fetchPrevious: false }
-);
 
-// If batch cancelled, add exit info to players
-Batches.after.update(
-  function(userId, { _id: batchId, status }, fieldNames, modifier, options) {
-    if (!fieldNames.includes("status")) {
+    if (status !== "cancelled") {
       return;
     }
 
-    if (status === "cancelled") {
-      const games = Games.find({ batchId }).fetch();
-      const gameLobbies = GameLobbies.find({ batchId }).fetch();
-      const gplayerIds = _.flatten(_.pluck(games, "playerIds"));
-      const glplayerIds = _.flatten(_.pluck(gameLobbies, "playerIds"));
-      const playerIds = _.union(gplayerIds, glplayerIds);
-      Players.update(
-        { _id: { $in: playerIds } },
-        { $set: { exitStatus: "gameCancelled", exitAt: new Date() } },
-        { multi: true }
-      );
+    const games = Games.find({ batchId }).fetch();
+    const gplayerIds = _.flatten(_.pluck(games, "playerIds"));
+
+    Players.update(
+      { _id: { $in: gplayerIds }, exitAt: { $exists: false } },
+      { $set: { exitStatus: "gameCancelled", exitAt: new Date() } },
+      { multi: true }
+    );
+
+    const gameLobbies = GameLobbies.find({
+      batchId,
+      gameId: { $exists: false }
+    }).fetch();
+
+    if (gameLobbies.length === 0) {
+      return;
     }
+
+    const glplayerIds = _.flatten(_.pluck(gameLobbies, "queuedPlayerIds"));
+    const players = Players.find({
```

Comment:
I guess we could check glplayerIds.length > 0 here. The game lobbies might all have 0 players. I think the Players.find will just return nothing, though I don't know, maybe it will throw. 👯

Reply:
Tested, does not throw 👍